examples/micropython/main.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2019 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       micropython example application
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "thread.h"
+
+#include "micropython.h"
+#include "py/stackctrl.h"
+#include "lib/utils/pyexec.h"
+
+#include "blob/boot.py.h"
+
+static char mp_heap[MP_RIOT_HEAPSIZE];
+
+int main(void)
+{
+    int coldboot = 1;
+
+    /* let MicroPython know the top of this thread's stack */
```

Comment:
Could the following code be moved to a shared function implemented in the package ?

Reply:
yes. at the moment, the example contains a little more code than needed, some of it could be factored out into a shared function. But we don't know yet how MicroPython is gonna be used with RIOT. Will modules use it? Or will it be used for applications?

I want to work out those details as we go...